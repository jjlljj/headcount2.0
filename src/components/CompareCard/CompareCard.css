.compare-card {
  text-align: center;
}

.compare-card h2 {
  font-size: 24px;
  font-weight: bold;
  margin: 20px 15px 5px;
}

.compare-card h3 {
  font-size: 18px;
  font-weight: bold;
  margin: 15px 15px 10px;
}

.compare-card h4 {
  font-size: 22px;
  font-weight: bold;
  margin: 15px 15px 10px;
}