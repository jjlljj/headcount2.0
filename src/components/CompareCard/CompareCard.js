import React from 'react';
import { oneOfType, string, object } from 'prop-types';
import './CompareCard.css';

const CompareCard = ({ card }) => {
  const cardKeys = Object.keys(card);
  const { compared } = card;

  return (
    <div className="data-card compare-card">
      <h2>Compared Districts</h2>
      <div>
        <h3>{cardKeys[0]}</h3>
        <p> &#8678; {card[cardKeys[0]]}</p>
      </div>  
      <div>
        <h4>Ratio</h4>
        <p>{compared}</p>
      </div>
      <div>
        <h3>{cardKeys[1]}</h3>
        <p>{card[cardKeys[1]]} &#8680;</p>
      </div>  
    </div>
  );
};

CompareCard.propTypes = {
  card: oneOfType( [object, string] )
};

export default CompareCard;
