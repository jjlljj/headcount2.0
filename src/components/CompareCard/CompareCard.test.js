/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import CompareCard from './CompareCard';

describe('CompareCard', () => {

  it('should exist', () => {
    const comparison = { "Denver, Co": 4321, "Boston, Ma": 1234, compared: .609 }
    const renderedComponent = shallow(<CompareCard card={comparison}/>)

    expect(renderedComponent).toBeDefined()
  })

  it('should should match its snapshot', () => {
    const comparison = { "Denver, Co": 4321, "Boston, Ma": 1234, compared: .609 }
    const renderedComponent = shallow(<CompareCard card={comparison}/>)

    expect(renderedComponent).toMatchSnapshot()
  })

  
});