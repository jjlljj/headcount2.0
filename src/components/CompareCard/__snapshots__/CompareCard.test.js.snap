// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCard should should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    card={
        Object {
            "Boston, Ma": 1234,
            "Denver, Co": 4321,
            "compared": 0.609,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Compared Districts
</h2>,
        <div>
          <h3>
                    Denver, Co
          </h3>
          <p>
                     ⇦ 
                    4321
          </p>
</div>,
        <div>
          <h4>
                    Ratio
          </h4>
          <p>
                    0.609
          </p>
</div>,
        <div>
          <h3>
                    Boston, Ma
          </h3>
          <p>
                    1234
                     ⇨
          </p>
</div>,
      ],
      "className": "data-card compare-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Compared Districts",
        },
        "ref": null,
        "rendered": "Compared Districts",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Denver, Co
</h3>,
            <p>
               ⇦ 
              4321
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Denver, Co",
            },
            "ref": null,
            "rendered": "Denver, Co",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ⇦ ",
                4321,
              ],
            },
            "ref": null,
            "rendered": Array [
              " ⇦ ",
              4321,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Ratio
</h4>,
            <p>
              0.609
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ratio",
            },
            "ref": null,
            "rendered": "Ratio",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0.609,
            },
            "ref": null,
            "rendered": 0.609,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Boston, Ma
</h3>,
            <p>
              1234
               ⇨
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Boston, Ma",
            },
            "ref": null,
            "rendered": "Boston, Ma",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                1234,
                " ⇨",
              ],
            },
            "ref": null,
            "rendered": Array [
              1234,
              " ⇨",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Compared Districts
</h2>,
          <div>
            <h3>
                        Denver, Co
            </h3>
            <p>
                         ⇦ 
                        4321
            </p>
</div>,
          <div>
            <h4>
                        Ratio
            </h4>
            <p>
                        0.609
            </p>
</div>,
          <div>
            <h3>
                        Boston, Ma
            </h3>
            <p>
                        1234
                         ⇨
            </p>
</div>,
        ],
        "className": "data-card compare-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Compared Districts",
          },
          "ref": null,
          "rendered": "Compared Districts",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Denver, Co
</h3>,
              <p>
                 ⇦ 
                4321
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Denver, Co",
              },
              "ref": null,
              "rendered": "Denver, Co",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ⇦ ",
                  4321,
                ],
              },
              "ref": null,
              "rendered": Array [
                " ⇦ ",
                4321,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Ratio
</h4>,
              <p>
                0.609
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ratio",
              },
              "ref": null,
              "rendered": "Ratio",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0.609,
              },
              "ref": null,
              "rendered": 0.609,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Boston, Ma
</h3>,
              <p>
                1234
                 ⇨
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Boston, Ma",
              },
              "ref": null,
              "rendered": "Boston, Ma",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  1234,
                  " ⇨",
                ],
              },
              "ref": null,
              "rendered": Array [
                1234,
                " ⇨",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
